enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -8813,15 +8884,40 @@ class AdjointGenerator
           IRBuilder<> Builder2(&call);
           getForwardBuilder(Builder2);
           auto origfree = orig->getArgOperand(0);
+          auto newfree = gutils->getNewFromOriginal(orig->getArgOperand(0));
           auto tofree = gutils->invertPointerM(origfree, Builder2);
-          if (tofree != origfree) {
+          auto newcall = gutils->getNewFromOriginal(orig);
+
+          auto rule = [&](Value *tofree) {
+            Value *ptrDiff = Builder2.CreatePtrDiff(tofree, newfree);
```

Comment:
Why not an icmp ne?

Reply:
> Why not an icmp ne?
`CreateIsNotNull` felt shorter...


```
  Value *CreateIsNotNull(Value *Arg, const Twine &Name = "") {
    return CreateICmpNE(Arg, Constant::getNullValue(Arg->getType()),
                        Name);
  }
```

Reply:
https://llvm.org/docs/LangRef.html#icmp-instruction

It should accept pointer types

Reply:
you are right.